scrc/dataset_creation/criticality_dataset_creator.py

```diff
@@ -1,93 +1,359 @@
-        origin_chambers = list(supreme_court_df.origin_chamber.unique())
-        self.logger.info(f"Found supreme court rulings with references to lower court rulings "
-                         f"from chambers {origin_chambers}")
-        return origin_chambers, supreme_court_df
+            citations = ast.literal_eval(citations_as_string)  # parse dict string to dict again
+            for citation in citations:
+                try:
+                    cit = citation['text']
+                    citation_type = citation['name']
+                    cit = ' '.join(cit.split())  # remove multiple whitespaces inside
+                    if citation_type == "ruling":
+                        cited_file = self.get_file_number(cit)
+                        cits.append(cited_file)
+                    elif citation_type == "law":
+                        pass
+                    else:
+                        raise ValueError("type of citation must be 'rulings' or 'law")
+                except ValueError as ve:
+                    self.logger.info("citation has invalid syntax")
+                    self.logger.info(citation)
+                    continue
+        except ValueError as ve:
+            self.logger.info("citations could not be extracted to dict")
+            self.logger.info(citations_as_string)
+        if cits:  # only return something if we actually have citations
+            return cits
+
+    def get_file_number(self, citation):
+        """
+        find for each citation string the matching citation from the start of bge (first page)
+        :param citation:         citation as string as found in text
+        :return:                 RulingCitation always in German
+        """
+        # handle citation always in German
+        found_citation = RulingCitation(citation, 'de')
+        if str(found_citation) in self.available_bges:
+            return found_citation
+        else:
+            # find closest bge with smaller page_number
+            year = found_citation.year
+            volume = found_citation.volume
+            page_number = found_citation.page_number
+            new_page_number = -1
+            for match in self.available_bges:
+                if f"BGE {year} {volume}" in match:
+                    tmp = RulingCitation(match, 'de')
+                    if new_page_number < tmp.page_number <= page_number:
+                        new_page_number = tmp.page_number
+            return RulingCitation(f"{year} {volume} {new_page_number}", 'de')
+
+    def save_huggingface_dataset(self, splits, feature_col_folder):
+        """
+        save data as huggingface dataset with columns: 'id', 'year': year, 'language',
+        region', 'canton', 'legal area', 'bge_label', 'considerations' and 'facts'
+        :param splits:                  specifying splits of dataset
+        :param feature_col_folder:      name of folder
+        """
```

Comment:
Is it not possible to use the generic save_huggingface_dataset function from the dataset_creator here?